generate_engagement_analysis_files.py

```diff
@@ -0,0 +1,109 @@
+from collections import OrderedDict
+import argparse
+import json
+import csv
+
+from core_data_modules.logging import Logger
+
+from src.lib import PipelineConfiguration
+
+Logger.set_project_name("IMAQAL")
+log = Logger(__name__)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Generates engagement data stored in multiple CSV sheets for analysis"
+                                                 " and visualization")
+
+    parser.add_argument("user", help="User launching this program")
+    parser.add_argument("pipeline_configuration_file_path", metavar="pipeline-configuration-file",
+                        help="Path to the pipeline configuration json file"),
+    parser.add_argument("demog_map_json_input_dir", metavar="demog-map-json-input-dir",
+                        help="Path to a directory to read per episode demog map .json files ")
+    parser.add_argument("engagement_csv_output_dir", metavar="engagement-csv-output-dir",
+                        help="Directory to write engagement CSV files to")
+
+
+    args = parser.parse_args()
+
+    user = args.user
+    pipeline_configuration_file_path = args.pipeline_configuration_file_path
+    demog_map_json_input_dir = args.demog_map_json_input_dir
+    engagement_csv_output_dir = args.engagement_csv_output_dir
+
+    engagement_map = {} # Participants shows and demogs data
```

Comment:
```suggestion
    engagement_map = {}  # Participants shows and demogs data
```
Think PyCharm will complain if you don't do this?

Reply:
Can you make this comment more descriptive? It's not clear when reading the file what this is a map from and to

Reply:
updated thanks!